XenAdmin/ConsoleView/VNCGraphicsClient.cs

```diff
@@ -248,8 +248,9 @@ private void vncStream_ConnectionSuccess(object sender, EventArgs e)
         private void OnError(object sender, Exception e)
         {
             Program.AssertOffEventThread();
-            System.Diagnostics.Trace.Assert(sender == this.vncStream);
-
+            if (sender != vncStream)

```

Comment:
I suggest retaining the Assert (perhaps making it into a Debug assert) but keeping the new "return" too.